packages/nx/src/daemon/server/project-graph-listener-sockets.ts

```diff
@@ -0,0 +1,35 @@
+import { Socket } from 'net';
+import { ProjectGraph } from '../../config/project-graph';
+import { ConfigurationSourceMaps } from '../../project-graph/utils/project-configuration-utils';
+import { handleResult } from './server';
+
+export let registeredProjectGraphListenerSockets: Socket[] = [];
+
+export function removeRegisteredProjectGraphListenerSocket(socket: Socket) {
+  registeredProjectGraphListenerSockets =
+    registeredProjectGraphListenerSockets.filter((s) => s !== socket);
+}
+
+export function hasRegisteredProjectGraphListenerSockets() {
+  return registeredProjectGraphListenerSockets.length > 0;
+}
+
+export async function notifyProjectGraphListenerSockets(
+  projectGraph: ProjectGraph,
+  sourceMaps: ConfigurationSourceMaps
+) {
+  if (!hasRegisteredProjectGraphListenerSockets()) {
+    return;
+  }
+
+  await Promise.all(
+    registeredProjectGraphListenerSockets.map((socket) =>
+      handleResult(socket, 'PROJECT_GRAPH_UPDATED', () =>
```

Comment:
Lets add these messages to `message-types`, gives a bit more type safety but is admittedly a bit more cruft